src/hooks/World.py

```diff
@@ -82,6 +83,15 @@ def after_create_regions(world: World, multiworld: MultiWorld, player: int):
 
 # The item pool before starting items are processed, in case you want to see the raw item pool at that stage
 def before_create_items_starting(item_pool: list, world: World, multiworld: MultiWorld, player: int) -> list:
+    itemNamesToRemove = []
+    for item in item_pool:
+        name = item.name
+        if "level" in item_name_to_item[name].keys():
+            if item_name_to_item[name]["level"] > get_option_value(multiworld, player, "level_cap"):
+                itemNamesToRemove.append(name)
+    for item in item_pool[:]:
+        if item.name in itemNamesToRemove and item.player == player:
+            item_pool.remove(item)

```

Comment:
This entire section isn't needed.  It's already done in https://github.com/silasary/APxiv/blob/master/src/hooks/World.py#L118-L122